lib/ansible/modules/storage/emc/emc_vplex_storagevolume.py

```diff
@@ -0,0 +1,183 @@
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = r'''
+---
+module: emc_vplex_storagevolume
+version_added: '2.9'
+short_description: Create storage-volume on EMC VPLEX Storage Array
+description:
+- Create storage-volume with verification of existence
+author:
+- Hiroyuki Wakabayashi (@hwakabh)
+options:
+    vplex_ip_address:
+        description:
+            - IP Address of VPLEX management server to create storage-volume
+        required: true
+    vplex_username:
+        description:
+            - Username of VPLEX management server to create storage-volume
+        required: true
+    vplex_password:
+        description:
+            - Password of VPLEX management server to create storage-volume
+        required: true
+    vplex_serialnum:
+        description:
+            - S/N of VPLEX to create storage-volume
+        required: true
+    volume_name:
+        description:
+            - Name of LUN to encapsulate in VPLEX
+        required: true
+    vpd_id:
+        description:
+            - WWN of LUN to encapsulate in VPLEX
+        required: true
+    array_name:
+        description:
+            - Name of backend storage-array with the name exposed from VPLEX
+        required: true
+'''
+
+EXAMPLES = r'''
+- name: Create storage-volume on VPLEX
+  emc_vplex_storagevolume:
+    vplex_ip_address: 10.10.10.2
+    vplex_username: service
+    vplex_password: password
+    vplex_serialnum: CKM00120900782
+    volume_name: lun_name_backend
+    vpd_id: VPD83T3:600601603a20310031ff7e125e33e911
+    array_name: EMC-CLARiiON-CKM00121900536
+'''
+
+RETURN = r'''
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.storage.emc.emc_vplex import VPLEX
+
+import logging
+import os
+
+logger = logging.getLogger(__name__)
+
+
+# storage-volumes
+def rediscover_storage_array(vplex, array_name):
+    uri = '/vplex/array+re-discover'
+    data = '{\"args\":\"--cluster cluster-1 --array ' + array_name + ' --force\"}'
+    response = vplex.https_post(urlsuffix=uri, data=data)
+    return response
+
+
+def get_storage_volume_info(vplex, name, is_all):
+    if is_all:
+        uri = '/vplex/clusters/cluster-1/storage-elements/storage-volumes/'
+    else:
+        uri = '/vplex/clusters/cluster-1/storage-elements/storage-volumes/' + name
+    response = vplex.https_get(urlsuffix=uri)
+    # logger.info(response)
+    return response
+
+
+def claim_storage_volume(vplex, volume_name, vpd_id):
+    claim_uri = '/vplex/storage-volume+claim'
+    claiming_data = '{\"args\":\"-n ' + volume_name + ' -d ' + vpd_id + ' -f\"}'
+    response = vplex.https_post(urlsuffix=claim_uri, data=claiming_data)
+    return response
+
+
+def main():
+    log_directory = "./logs"
+    if not os.path.exists(log_directory):
+        os.makedirs(log_directory)
```

Comment:
I may not be allowed to write to current directory. Maybe log path should be passed as parameter.